### PR TITLE
Use buildah instead of manifest-tool to create manifest lists

### PR DESCRIPTION
**Pull request by yashvardhannanavati**
Relates to CLOUDDST-8606

### CODE REVIEW

File: iib/workers/tasks/build.py [hunk 1]
```diff
@@ -116,45 +116,36 @@ def _create_and_push_manifest_list(request_id, arches):
...
-        manifest_yaml = os.path.abspath(os.path.join(temp_dir, 'manifest.yaml'))
-        with open(manifest_yaml, 'w+') as manifest_yaml_f:
-            manifest_yaml_f.write(
-                textwrap.dedent(
-                    f'''\
-                    image: {output_pull_spec}
-                    manifests:
-                    '''
-                )
+        for arch in sorted_arches:
```
Comment (pbortlov) [inline iib/workers/tasks/build.py:124]: would it be possible to use line 122 here? if you don't use sorted_aches anywhere else?

File: iib/workers/tasks/build.py [hunk 2]
```diff
@@ -116,45 +116,36 @@ def _create_and_push_manifest_list(request_id, arches):
...
-                manifest_yaml_f.write(
-                    textwrap.dedent(
-                        f'''\
-                        - image: {arch_pull_spec}
-                          platform:
-                            architecture: {arch}
-                            os: linux
-                        '''
-                    )
+            if arch == sorted_arches[0]:
```
Comment (pbortlov) [inline iib/workers/tasks/build.py:131]: Could you add a comment here?

File: iib/workers/tasks/build.py (hunk 1 above)
Comment (yashvardhannanavati) [reply iib/workers/tasks/build.py:124]: But I am using it on Line 130

File: iib/workers/tasks/build.py (hunk 2 above)
Comment (yashvardhannanavati) [reply iib/workers/tasks/build.py]: Done

File: iib/workers/tasks/build.py [hunk 3]
```diff
@@ -116,48 +115,36 @@ def _create_and_push_manifest_list(request_id, arches):
...
+    for arch in sorted_arches:
+        arch_pull_spec = _get_external_arch_pull_spec(request_id, arch, include_transport=True)
+        log.debug(
+            'Adding the manifest %s to the manifest list %s', arch_pull_spec, output_pull_spec
+        )
+        # Create the manifest list with the first arch and then add other arches to it using the
+        # buildah-manifest add command.
+        if arch == sorted_arches[0]:
+            run_cmd(
+                buildah_manifest_cmd + ['create', output_pull_spec, arch_pull_spec],
```
Comment (lcarva) [inline iib/workers/tasks/build.py:131]: You can call `create` with just the pull spec of the manifest list. Consider this pattern:
```python
run_cmd(
    buildah_manifest_cmd + ['create', output_pull_spec],
    exc_msg=f'Failed to create the manifest list locally: {output_pull_spec}',
)
...
for arch in sorted_arches:
...
  run_cmd(
       buildah_manifest_cmd + ['add', output_pull_spec, arch_pull_spec],
       exc_msg=(
          f'Failed to add {arch_pull_spec} to the' f' local manifest list: {output_pull_spec}'
       ),
  )
...
```

It has the benefit of not treating the first arch a special case.

File: iib/workers/tasks/build.py [hunk 4]
```diff
@@ -116,48 +115,36 @@ def _create_and_push_manifest_list(request_id, arches):
...
+                exc_msg=f'Failed to create the manifest list locally: {output_pull_spec}',
             )
-
+            continue
         run_cmd(
-            ['manifest-tool', 'push', 'from-spec', manifest_yaml],
-            exc_msg=f'Failed to push the manifest list to {output_pull_spec}',
+            buildah_manifest_cmd + ['add', output_pull_spec, arch_pull_spec],
```
Comment (lcarva) [inline iib/workers/tasks/build.py:136]: Do we need to use `--os` or `--arch` here?

**Review by lcarva — COMMENTED** [review]
Let's also remove all references to `manifest-tool`, e.g. in the Dockerfile.

File: iib/workers/tasks/build.py (hunk 4 above)
Comment (yashvardhannanavati) [reply iib/workers/tasks/build.py:136]: I don't think we need `--os` here and we build the image using `--arch` so the metadata on the image is already set correctly. I could add it if you think it's safe to do

File: iib/workers/tasks/build.py (hunk 3 above)
Comment (yashvardhannanavati) [reply iib/workers/tasks/build.py]: Done

File: iib/workers/tasks/build.py (hunk 4 above)
Comment (lcarva) [reply iib/workers/tasks/build.py:131]: If it generates the correct manifest, then there's no need for it. Have you inspected the raw manifest list created in the registry?

Comment (yashvardhannanavati) [reply iib/workers/tasks/build.py:131]: yes, I have inspected it and it looks correct

Comment (lcarva) [reply iib/workers/tasks/build.py:131]: Cool!

**Comment by lcarva** [comment]
> Let's also remove all references to `manifest-tool`, e.g. in the Dockerfile.

@yashvardhannanavati, did you see this ^?

**Review by lcarva — DISMISSED** [review]

**Comment by yashvardhannanavati** [comment]
> > Let's also remove all references to `manifest-tool`, e.g. in the Dockerfile.
> 
> @yashvardhannanavati, did you see this ^?

I hadn't seen it. I just added a commit removing it from docker-compose.

**Review by lcarva — APPROVED** [review]

**Review by lipoja — APPROVED** [review]

**Pull request merged**